js/runner_test.go

```diff
@@ -910,7 +910,15 @@ func TestVUIntegrationMetrics(t *testing.T) {
 	}
 }
 
+func GenerateTLSCertificateWithCA(t *testing.T, host string, notBefore time.Time, validFor time.Duration, caCert *x509.Certificate, caKey *rsa.PrivateKey) ([]byte, []byte) {
```

Comment:
```suggestion
func generateTLSCertificateWithCA(t *testing.T, host string, caCert *x509.Certificate, caKey *rsa.PrivateKey) ([]byte, []byte) {
```
- Do we need to have it exported? (In alternative, if we find a way to make it a bit simpler API and we have other tests that could benefit from it then we may consider moving to the `testutils` package.)
- It sounds like we always use the same time config so we may skip it from the API
- Do we need to define a CA? Could we use a pre-defined CA for all our internal tests? Like `k6ca.test.example`

Reply:
> Do we need to have it exported? 

I don't think so, as it's only used within this file.
I just kept it exporter because it already was, similar to `GetTestServerWithCertificate` function also present in that file, but I'm happy to make all these unexported.

> It sounds like we always use the same time config so we may skip it from the API
> Do we need to define a CA? Could we use a pre-defined CA for all our internal tests? Like k6ca.test.example

I'm not sure what you mean here. Do you mean the CA args?
Could you elaborate a bit more how would you have a pre-defined CA? 
Storing the CA cert/key as a global variable, and having a `sync.Once` to initialize it once? Or now?
Because I think in this case we need to generate two certificates (client & server) that are signed with the same CA cert & key.

Thanks! 🙏🏻

Reply:
>I'd feel like prematurely generalizing.

Yes, I feel the same if we don't have other occurrences. Then we can unexport them.

>I'm not sure what you mean here.

Sorry, I meant the host.

Reply:
> Sorry, I meant the host.

Okay! So, technically it's true that all the current calls to `GenerateTLSCertificateWithCA` use the `host` = `127.0.0.1` (in fact, I think it only really matters for the server one), but at the same time I feel like hardcoding that value would make that method less usable, as it's exactly the same as the former `GenerateTLSCertificate` but with custom CA cert & key, so `host` could be any value (the consumer needs, like with the former method).

In fact, the host for the CA is defined on the first call to `GenerateTLSCertificate`, and that method is used with different `host` along the code, so that function needs to remain with a parameterized host.

So, that being said, unless you want to suggest something different specifically (please do, if you want!), I'd prefer to leave it as is, because I don't see any good way to do it.

Reply:
And yes, I'm going to turn these methods into unexported ones! 🙏🏻